roles/testnode/README.rst

```diff
@@ -0,0 +1,206 @@
+Testnode
+========
+
+This role is used to configure a node for ceph testing using teuthology_ and ceph-qa-suite_.
+It will manage the necessary groups, users and configuration needed for teuthology to connect to and use the node.
+It also installs a number of packages needed for tasks in ceph-qa-suite and teuthology.
+
+The following distros are supported:
+
+- RHEL 6.X
+- RHEL 7.X
+- Centos 6.X
+- Centos 7.x
+- Fedora 20
+- Debian Wheezy
+- Ubuntu Precise
+- Ubuntu Trusty
+- Ubuntu Vivid
+
+**NOTE:** This role was first created as a port of ceph-qa-chef_.
+
+Usage
++++++
+
+The testnode role is primarily used by the ``testnodes.yml`` playbook.  This playbook is run by cobbler during
+bare-metal imaging to prepare a node for testing and it also used by Teuthology during test runs to ensure the config
+is correct before testing.
+
+**NOTE:** ``testnodes.yml`` is limited to run against hosts in the ``testnodes`` group by the ``hosts`` key in the playbook.
+
+Variables
++++++++++
+
+Available variables are listed below, along with default values (see ``roles/testnode/defaults/main.yml``). This role also
+allows for variables to be defined per package type (apt, yum), distro, distro major version and distro version.
+These overrides are included by ``tasks/vars.yml`` and the specific var files live in ``vars/``.
+
+The host to use as a package mirror::
+
+    mirror_host: apt-mirror.front.sepia.ceph.com
+
+The host to use as a github mirror::
+
+    git_mirror_host: git.ceph.com
+
+This host is used by teuthology to download ceph packages and will be given higher priority on apt systems::
+
+    gitbuilder_host: gitbuilder.ceph.com
+
+The mirror to download and install ``pip`` from::
+
+    pip_mirror_url: "http://{{ mirror_host }}/pypi/simple"
```

Comment:
Currently this must be defined.  I'd like mirror_host, gitbuilder_host and
pip_mirror_url to all noop eventually if they aren't provided. Then we'd
change all the defaults here to empty strings.

On Wed, Jul 15, 2015 at 1:15 PM, Dan Mick <notifications@github.com> wrote:

> In roles/testnode/README.rst
> <https://github.com/ceph/ceph-cm-ansible/pull/75#discussion_r34710571>:
>
> > +
> > +The host to use as a package mirror::
> > +
> > +    mirror_host: apt-mirror.front.sepia.ceph.com
> > +
> > +The host to use as a github mirror::
> > +
> > +    git_mirror_host: git.ceph.com
> > +
> > +This host is used by teuthology to download ceph packages and will be given higher priority on apt systems::
> > +
> > +    gitbuilder_host: gitbuilder.ceph.com
> > +
> > +The mirror to download and install ``pip`` from::
> > +
> > +    pip_mirror_url: "http://{{ mirror_host }}/pypi/simple"
>
> Will this work if mirror_host is undefined (i.e. "don't mirror it")?
> Should it?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ceph/ceph-cm-ansible/pull/75/files#r34710571>.
>